### DISCUSSION TITLE
tasks -> send to terminal?

### DISCUSSION THREAD

**Discussion by cs2018ncsa** [zed-industries/zed#14058] [Feature Requests]
I've been playing with tasks for a bit. I am trying to get something similar to the send-to-terminal capability in VSCode. i have a terminal open w/ a psql connection to a postgres database. i'd like to highlight a command and sent that to the terminal for execution. i _think_ i can do that as a task. are there any examples of this anywhere?

**Comment by tepavcevic** [comment]
Closest thing to that could be configuring task for that command e.g.:
```json
{
    "label": "Example command",
    "command": "do this command",
    "use_new_terminal": true,
    "reveal": "always"
  }
```
and then to have a keybinding in your keymap config to execute the task:
```json
"ctrl-whatever": [
        "task::Spawn",
        {
          "task_name": "Example command"
        }
      ]
```
I use it like this for my git workflow with `gitui`

**Comment by cs2018ncsa** [comment]
@tepavcevic thanks for response!

I can get it to work if i launch psql in the "command" with $ZED_SELECTED_TEXT in "args" but i would like to execute commands in already connected psql session. so i don't think any command (not possible). i tried the following with no success either.

also, i set "user_new_terminal" to false and it still opens a new terminal.

  {
    "label": "Run SQL selection 2",
    "command": "$ZED_SELECTED_TEXT;",
    "use_new_terminal": false
  }

i wish there was more documentation on this.

**Reply by tepavcevic** [reply]
I see, there is no way for a task to forward that input to the running terminal session. Unfortunately I am not aware of a way to do exactly that, but perhaps someone else can be of more help?

**Comment by mcabbott** [comment]
Also requested in #8674, which was closed as a dup of bigger issue #8344 . Would be nice to have a way to do this!

**Reply by mmyrte** [reply]
Both of those items have been converted to discussions - currently, there is a PR being worked on / discussed at #42467. Since that actually uses LSP for scoping, it should be more akin to the "specialised REPL-IDE" experience that I know from various data science tools and languages.

**Comment by dballaelliott** [comment]
A workaround using `workspace::SendKeystrokes` seems to be working for me 

I added this to the global [context](https://zed.dev/docs/key-bindings#contexts) in my `keymap.json`
```
    // keybinding to paste the selected text into the terminal
      "cmd-enter": [
        "workspace::SendKeystrokes",
        "cmd-c ctrl-` cmd-v enter ctrl-`"
      ]
```

This seems a bit hacky and relies on the default `Workspace` binding 

```
     "ctrl-`": "terminal_panel::ToggleFocus",
```

This copies the whatever text is selected, switches to terminal, pastes, hits `enter` and then switches back to editor.

**Reply by afrendeiro** [reply]
Thanks a lot!
On Linux:
```json
"ctrl-enter": [
  "workspace::SendKeystrokes",
  "ctrl-c ctrl-` ctrl-shift-v enter ctrl-` end down"
]
```
This already works very nicely, but unfortunately does not deal with multi-line statements and many other more complex cases.
Support out-of-the box would be amazing.

**Comment by choucavalier** [comment]
agreed this is missing and would be great for those like me accustomed to the vim-slime workflow

**Comment by marianoarnaiz** [comment]
Hi. I am new to ZED. Where can I try this to add the capabilities to ZED? I am working on a MAC

**Comment by syepes** [comment]
I had this working in mac "cmd-c ctrl-\` cmd-v enter ctrl-\` end down", and its no longer working with Zed 0.204.5, anyone else noticed this?

**Reply by syepes** [reply]
For anyone else looking for this, below is a new version that works
```
  {
    "context": "Editor",
    "bindings": {
      "shift-enter": [
        "workspace::SendKeystrokes",
        "cmd-c ctrl-` cmd-v enter cmd-shift-e cmd-shift-e down"
      ]
    }
  }
```

**Reply by dstmu** [reply]
Thanks!!
"Send to open terminal" really has to be added, esp. since inline execution of code for many languages (e.g., R) is pretty substandard (does not deal with project-specific setups such as  conda or pixi etc).

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
